saleor/tests/e2e/checkout/test_checkout_products_on_fixed_promotion.py

```diff
@@ -0,0 +1,222 @@
+import pytest
+
+from ..channel.utils import create_channel
+from ..product.utils import (
+    create_category,
+    create_product,
+    create_product_channel_listing,
+    create_product_type,
+    create_product_variant,
+    create_product_variant_channel_listing,
+)
+from ..promotions.utils import create_promotion, create_promotion_rule
+from ..shipping_zone.utils import (
+    create_shipping_method,
+    create_shipping_method_channel_listing,
+    create_shipping_zone,
+)
+from ..utils import assign_permissions
+from ..warehouse.utils import create_warehouse, update_warehouse
+from .utils import (
+    checkout_complete,
+    checkout_create,
+    checkout_delivery_method_update,
+    checkout_dummy_payment_create,
+)
+
+
+def prepare_product(
```

Comment:
I see that we are defining such method quite often in different test files, cannot we reuse it and just for example here extend it with the promotion data?

Reply:
This is something we want to address in a separate PR